e2e/puppeteer-custom-environment.ts

```diff
@@ -18,28 +15,28 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
   }
 
   // Take a screenshot right after failure
-  async handleTestEvent(event, state) {
+  async handleTestEvent(event) {
     switch (event.name) {
-    case 'test_fn_failure':
-      if (state.currentlyRunningTest.invocations > retryTimes) {
-        console.error(`Failed test:  "${event.test.name}"`);
-        const testName = fp.startCase(state.currentlyRunningTest.name).replace(/[^A-Z0-9]+/ig, '');
-        const screenshotDir = 'logs/screenshot';
-        await fs.ensureDir(screenshotDir);
+      // case 'test_fn_failure':
+      // case 'hook_failure':
+      case 'test_done':
+        if (event.test.errors.length > 0) {
+          console.error(`handleTestEvent case: ${event.name}`);
+          console.error(`Failed test:  "${event.test.name}"`);
+          const testName = event.test.name.replace(/\W/g, '-');
+          const screenshotDir = 'logs/screenshot';
+          await fs.ensureDir(screenshotDir);
           // move create-filename to helper.ts
```

Comment:
nit: this comment seems out of place, especially after the refactor. Maybe remove it?